src/windowmanagement/WindowLayout/WindowLayoutProvider.tsx

```diff
@@ -0,0 +1,104 @@
+import { createContext, Ref, useContext,useRef } from 'react';
+import { ScrollArea } from '@mantine/core';
+import DockLayout, { BoxData, PanelData, TabData } from 'rc-dock';
+
+import { WindowLayoutOptions } from './WindowLayout';
+
+export interface ProviderProps {
+  ref: Ref<DockLayout>;
+  addWindow: (component: JSX.Element, options: WindowLayoutOptions) => void;
+}
+
+export const WindowManagerContext = createContext<ProviderProps | null>(null);
+
+export function WindowManagerProvider({ children }: { children: React.ReactNode }) {
+  const rcDocRef = useRef<DockLayout>(null);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function isPanelDataInstance(obj: any) {
+    return obj && Array.isArray(obj.tabs);
```

Comment:
Not sure if we tried this the first time, but shouldn't we be able to do `obj: object`?

Reply:
Fixed it